### PR TITLE
rename ure, ujson, and uerrno to re, json, and errno

### PR DESCRIPTION
**Pull request by dhalbert**
Use original CPython names for `ure`, `ujson` and `uerrno`. These libraries are all strict subsets.

### CODE REVIEW

**Comment by jerryneedell** [comment]
@dhalbert will both modules continue to be available? If not some libraries will be broken.

**Comment by dhalbert** [comment]
@jerryneedell I intend to fix the broken libraries with try-catch on the imports. There seem to be very few to fix(like one or two). There is probably also Learn Guide code to fix.

**Comment by makermelissa** [comment]
Not sure if it’s on your list, but ESP32SPI uses ujson.

**Comment by dhalbert** [comment]
Thanks - my bundle was out of date.

File: extmod/modure.c [hunk 1]
```diff
@@ -405,7 +405,7 @@ STATIC MP_DEFINE_CONST_DICT(re_locals_dict, re_locals_dict_table);
 
 STATIC const mp_obj_type_t re_type = {
     { &mp_type_type },
-    .name = MP_QSTR_ure,
+    .name = MP_QSTR_re,
```
Comment (tannewt) [inline extmod/modure.c]: #if CIRCUITPY this too if you are doing it elsewhere.

Comment (dhalbert) [reply extmod/modure.c]: Oops, yes.

**Pull request merged**